.github/workflows/main.yml

```diff
@@ -35,18 +36,64 @@ jobs:
 
       - name: Save docker image
         run: |
-          docker save research-template | gzip > /tmp/research-template.tar.gz
+          docker save research-template | pigz --fast > /tmp/research-template.tar.gz
 
       - name: Upload docker image
         uses: actions/upload-artifact@v4
         with:
             name: research-template-image
             path: /tmp/research-template.tar.gz
+            # Disable compression; the file is already compressed
+            compression-level: 0
 
-  publish:
+  test-devcontainer:
     needs: [build-and-test]
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Download Docker image
+      uses: actions/download-artifact@v4
+      with:
+        name: research-template-image
+        path: /tmp/image
+
+    - name: Import Docker image
+      run: docker load --input /tmp/image/research-template.tar.gz
+
+    - name: Tag Docker image for use with dev container
+      run: docker tag "$IMAGE_NAME" "$PUBLIC_IMAGE_NAME:$IMAGE_VERSION"
+
+    - name: Checkout research template temporary devcontainer dev branch
+      uses: actions/checkout@v4
+      with:
+        repository: opensafely/research-template
 
-    runs-on: ubuntu-22.04
+    - name: Checkout research-template-docker repository in subdirectory
+      uses: actions/checkout@v4
+      with:
+        path: 'research-template/research-template-docker'
+
+    - name: Install demjson package for jsonlint
+      # Necessary because jq doesn't yet have an option to strip comments.
+      # See https://github.com/jqlang/jq/issues/1571
+      run: |
+        sudo apt-get update
+        sudo apt-get install python3-demjson
+
+    - name: Check that we have correct version of image in devcontainer.json
+      run: |
+        devcontainer_image=$(jsonlint --allow comments --format-compactly ".devcontainer/devcontainer.json" | jq --raw-output .image)
+        test "$devcontainer_image" = "$PUBLIC_IMAGE_NAME:$IMAGE_VERSION"
+
+    - name: Build and run dev container task
+      uses: devcontainers/ci@v0.3
```

Comment:
Forgot to pin this to a commit, instead of a tag that can move, so should do that before merging.

Reply:
Ah, I remember that I didn't actually forget, but I left this for another reason.

The `devcontainers/ci` repository is slightly peculiar here. If you select the commit associated with a release, then you end up on a commit that's not on a branch.

[Example commit for the current release v0.3.1900000349](https://github.com/devcontainers/ci/commit/a56d055efecd725e8cfe370543b6071b79989cc8).

I guess that's OK, but I'm not sure how that interacts with Dependabot updates, so I'll check that.

Reply:
Tested this out in a repository that had an older version of this action in a workflow, and it updates OK, so will pin the commit.